### ISSUE TITLE
coqsplit.ml: error with string not bytes

### ISSUE THREAD

**Issue by spl** [plclub/metalib#14]
I ran into this error:

```
$ cd Stlc; make
ocamlc.opt coqsplit.ml -o coqsplit
File "coqsplit.ml", line 295, characters 15-28:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "coqsplit.ml", line 420, characters 34-35:
Error: This expression has type bytes but an expression was expected of type
         string
make: *** [coqsplit] Error 2
```

I'm guessing this is a problem with a newer OCaml:

```
$ ocamlc.opt --version
4.06.0
```

Perhaps it's due the [change to immutable strings in OCaml-4.02](http://blog.camlcity.org/blog/bytes1.html)?

I “fixed” it with the following patch, but I have no confidence that it's correct (I'm not using the output of `coqsplit.ml`):

```diff
diff --git a/Stlc/coqsplit.ml b/Stlc/coqsplit.ml
index 7a5a9b4..00b3675 100644
--- a/Stlc/coqsplit.ml
+++ b/Stlc/coqsplit.ml
@@ -291,10 +291,7 @@ let spec =
 (* tools to read and process tags on the input strings *)

 let readchan chan =
-  let nbytes = in_channel_length chan in
-  let string = String.create nbytes in
-  really_input chan string 0 nbytes;
-  string
+  really_input_string chan 0

 let findsubstring s1 s2 =
   let l1 = String.length s1 in
```

**Comment by sweirich** [comment]
There is also a fix for this issue in PR #15 (which I don't want to merge yet).    I'm also not using the output of coqsplit so I don't know if either suffice.

**Comment by sweirich** [comment]
Fixed by c716013d14f04b399f53afc34723ab7bcb4439cc
